platforms/documentation/docs/src/docs/userguide/releases/feature_lifecycle.adoc

```diff
@@ -133,3 +133,197 @@ As such, each major Gradle release causes:
 
 * The previous major version becomes maintenance only. It will only receive critical bug fixes and security fixes.
 * The major version before the previous one to become end-of-life (EOL), and that release line will not receive any new fixes.
+
+== Public Gradle APIs
+
+Gradle provides many public APIs that allow developers to interact with various build system components.
+
+Any class name containing `Internal`, or that is located in a package containing the segment `.internal.` is by definition not part of the public Gradle API and should not be relied upon for stable usage.
+
+The following table summarizes public Gradle APIs and their common use cases.
+This is *not* an exhaustive list:
+
+[cols="1,1,2"]
+|===
+| API Package | Description | Common Use Cases
+
+| *`org.gradle.api`*
+| Core Gradle API
+| - Defines key interfaces like `Project`, `Task`, and `Plugin` +
+- Used for configuring projects and tasks
+
+| *`org.gradle.authentication`*
+| Authentication API
+| - Handles authentication for repositories and services +
+- Supports username/password, OAuth, and other mechanisms
+
+| *`org.gradle.build`*
+| Build Lifecycle API
+| - Manages build phases and execution +
+- Useful for controlling the order of tasks
+
+| *`org.gradle.buildinit`*
+| Build Init API
+| - Used to generate new Gradle projects +
+- Supports bootstrapping various project types
+
+| *`org.gradle.caching`*
+| Build Cache API
+| - Provides mechanisms for storing and reusing build outputs +
+- Used for incremental builds and CI optimization
+
+| *`org.gradle.concurrent`*
+| Concurrent Execution API
+| - Manages parallel task execution +
+- Helps optimize Gradle build performance
+
+| *`org.gradle.deployment`*
+| Deployment API
+| - Supports managing and automating application deployments +
+- Useful for CI/CD pipelines
```

Comment:
🤔  I don't think this is useful for CI pipelines. This is more about deploying your application locally and running tests against it or doing hot reload. Currently, there is also no public API in the package, everything is internal.